### PR TITLE
rewrite class to func comp

### PR DESCRIPTION
**Pull request by davidbes**
I really like your repository and I found it quite helpful. 
However I had to adjust it to fit my needs so I rewrote your repo from Class components to arrow function component that uses hooks instead of basic component lifecycle methods. I did that so I could find my way around the code more easily. It probably can be adjusted.
The changes I do not change your original code. They just add an option to use either default Pixelfy or PixelifyWithHooks component.

### CODE REVIEW

File: lib/elements/PixelifyHooks.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import React, { useEffect, useRef } from 'react';
+import paintPixels from '../helpers/paintPixels';
+
+const PixelifyHook = ({ src, width, height, pixelSize, centered, fillTransparencyColor }) => {
+	const canvasRef = useRef(null);
+
+	useEffect(() => {
+		const parsedPixelSize = parseInt(pixelSize, 10);
+		console.log(parsedPixelSize);
```
Comment (nikoferro) [inline lib/elements/PixelifyHooks.js]: hey, can you remove the console log? Thanks!

**Review by nikoferro — COMMENTED** [review]
Hey! Thanks for the PR, really appreciate you took the time. I need to take some time to review the code, so its not a promise it will get merged, but i ll take a look at it. Will be reviewing it during the next couple of days

File: lib/elements/PixelifyHooks.js (hunk 1 above)
Comment (davidbes) [reply lib/elements/PixelifyHooks.js]: Guess I missed one 👍

**Comment by davidbes** [comment]
One option to change would also be to make the image smaller and then resize it. That would mean, that it would take an image, paint it to canvas with smaller size, get image data back and redraw it on 2nd canvas with images original size. Just something to think about.

**Comment by nikoferro** [comment]
Closed this for now, since we first need tests before approaching such a big change. I wouldnt want to maintain two components doing the same thing (which is the case here, one w/ class, and one functional). Having said that, **i do really appreciate you contributing to the library**. Hopefully once we have tests in place, we can take a new look at this PR to migrate the component to functional and deprecate class usage.

**Pull request closed without merging**